src/lib.rs

```diff
@@ -33,33 +39,84 @@ use std::os::unix::prelude::*;
 use std::path::Path;
 
 use bytes::{Buf, BufMut};
-use futures::{Future, Poll, Async, Stream};
-use futures::sync::oneshot;
+use futures::{Async, Future, Poll, Stream};
+#[cfg(feature = "unstable-futures")]
+use futures2::task;
+use tokio_io::{AsyncRead, AsyncWrite};
 use iovec::IoVec;
-use tokio_core::reactor::{PollEvented, Handle};
-#[allow(deprecated)]
-use tokio_core::io::Io;
-use tokio_io::{IoStream, AsyncRead, AsyncWrite};
+use tokio_reactor::{Handle, PollEvented};
+use mio::Ready;
+
+#[cfg(feature = "unstable-futures")]
+macro_rules! try_ready2 {
+    ($e: expr) => {
+        match $e {
+            Ok($crate::futures2::Async::Ready(t)) => t,
+            Ok($crate::futures2::Async::Pending) => return Ok($crate::futures2::Async::Pending),
+            Err(e) => return Err(From::from(e)),
+        }
+    };
+}
 
 mod frame;
-pub use frame::{UnixDatagramFramed, UnixDatagramCodec};
+pub use frame::{UnixDatagramCodec, UnixDatagramFramed};
 mod ucred;
 pub use ucred::UCred;
 
-fn would_block() -> io::Error {
-    io::Error::new(io::ErrorKind::WouldBlock, "would block")
+#[cfg(feature = "unstable-futures")]
+fn lift_async<T>(old: futures::Async<T>) -> futures2::Async<T> {
+    match old {
+        futures::Async::Ready(x) => futures2::Async::Ready(x),
+        futures::Async::NotReady => futures2::Async::Pending,
+    }
+}
+
+/// Stream of listeners
+pub struct Incoming {
+    inner: UnixListener,
+}
+
+impl Stream for Incoming {
+    type Item = (UnixStream, SocketAddr);
+    type Error = io::Error;
+
+    fn poll(&mut self) -> Poll<Option<Self::Item>, io::Error> {
+        Ok(Some(try_nb!(self.inner.accept())).into())
+    }
+}
+
+#[cfg(feature = "unstable-futures")]
+impl futures2::Stream for Incoming {
+    type Item = (UnixStream, SocketAddr);
+    type Error = io::Error;
+
+    fn poll_next(
+        &mut self,
+        cx: &mut task::Context,
+    ) -> futures2::Poll<Option<Self::Item>, io::Error> {
+        Ok(Some(try_ready2!(self.inner.poll_accept2(cx))).into())
+    }
 }
 
 /// A Unix socket which can accept connections from other unix sockets.
 pub struct UnixListener {
     io: PollEvented<mio_uds::UnixListener>,
-    pending_accept: Option<oneshot::Receiver<io::Result<(UnixStream, SocketAddr)>>>,
+    handle: Handle,
 }
 
 impl UnixListener {
     /// Creates a new `UnixListener` bound to the specified path.
-    pub fn bind<P>(path: P, handle: &Handle) -> io::Result<UnixListener>
-        where P: AsRef<Path>
+    pub fn bind<P>(path: P) -> io::Result<UnixListener>
+    where
+        P: AsRef<Path>,
+    {
+        UnixListener::bind_handle(path.as_ref(), Handle::default())
+    }
+
+    /// Creates a new `UnixListener` bound to the specified path.
+    pub fn bind_handle<P>(path: P, handle: Handle) -> io::Result<UnixListener>
```

Comment:
This function can be removed.

Reply:
I was under the impression that there should both be a method that takes an explicit handle and one without (that grabs the default handle). Is that not the case anymore?